src/mahoji/commands/mix.ts

```diff
@@ -144,7 +146,21 @@ export const mixCommand: OSBMahojiCommand = {
 				maxTripLength
 			)}, try a lower quantity. The highest amount of ${itemName} you can mix is ${maxCanMix}.`;
 
+		const hasScroll = user.owns('Scroll of cleansing');
 		const finalCost = requiredItems.clone().multiply(quantity);
+
+		if (hasScroll) {
+			for (const [item, qty] of finalCost.items()) {
+				if (secondaries.includes(item.id)) {
+					const saved = Math.floor(calcPercentOfNum(10, qty));
+					finalCost.remove(item.id, saved);
+					const savedBank = new Bank();
+					savedBank.add(item.id, saved);
+					userStatsBankUpdate(user.id, 'cleansing_scroll_bank', savedBank);
```

Comment:
This is probably unsound (prone to overwriting updates due to not awaiting), and you should do it with just 1 call.

Possibly like:

```ts
const savedBank = new Bank();
for (const item of items) {
const savings = new Bank();
cost.remove(savings);
savedBank.add(savings);
}
await userStatsBankUpdate(user.id, 'cleansing_scroll_bank', savedBank);
```